Petar_Nedyalkov/task3.cpp

```diff
@@ -0,0 +1,92 @@
+#include <iostream>
+
+#pragma warning (disable : 4996)
+
+enum class Course {
+	first, second, third, fourth
+};
+
+struct Student
+{
+	double averageGrade;
+	int facultyNumber;
+	Course c;
+	char firstName[17];
+};
+
+void printStudent(const Student& toPrint)
+{
+	std::cout << toPrint.firstName << " " << toPrint.facultyNumber << " ";
+
+	switch (toPrint.c)
+	{
+	case Course::first: std::cout << "first"; break;
+	case Course::second: std::cout << "second"; break;
+	case Course::third: std::cout << "third"; break;
+	case Course::fourth: std::cout << "fourth"; break;
+		default:break;
+	};
+
+	std::cout <<" " << toPrint.averageGrade << std::endl;
+}
+
+bool initStudent(Student& toInit, char name[17], int facNumber, Course currentCourse, double grade)
```

Comment:
This hasn't really been covered yet (exceptions & error handling) but having a boolean return value to signal the success / failure of an operation (in this case - the initialization of an object) is not a good idea. Boolean values don't really mean much (if false nobody will be able to tell the tale of what exactly went wrong) and can easily be overlooked (e.g. if I don't do an explicit check about the given return value - I'll never now that an error occurred). Keep in mind that [some other mechanisms](https://www.tutorialspoint.com/cplusplus/cpp_exceptions_handling.htm) exist for doing just that :))